From c95da8545f36a5dd97703dcb13559ac6112589de Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Wed, 7 Aug 2019 18:26:51 +0800
Subject: [PATCH] Rule kernel_module_usb-storage_disabled add oval check and
 remediate script

For rule kernel_module_usb-storage_disabled add WRLinux1019 oval check and
remediate script.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                            |  8 ++
 .../oval/wrlinux1019.xml                           | 88 ++++++++++++++++++++++
 2 files changed, 96 insertions(+)
 create mode 100644 linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/bash/wrlinux1019.sh
 create mode 100644 linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/oval/wrlinux1019.xml

diff --git a/linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/bash/wrlinux1019.sh b/linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/bash/wrlinux1019.sh
new file mode 100644
index 0000000..333a5da
--- /dev/null
+++ b/linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/bash/wrlinux1019.sh
@@ -0,0 +1,8 @@
+# platform = multi_platform_wrlinux
+
+if grep --silent "^install usb-storage" /etc/modprobe.d/usb-storage.conf ; then
+	sed -i 's/^install usb-storage.*/install usb-storage /bin/true/g' /etc/modprobe.d/usb-storage.conf
+else
+	echo -e "\n# Disable per security requirements" >> /etc/modprobe.d/usb-storage.conf
+	echo "install usb-storage /bin/true" >> /etc/modprobe.d/usb-storage.conf
+fi
diff --git a/linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/oval/wrlinux1019.xml b/linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/oval/wrlinux1019.xml
new file mode 100644
index 0000000..b1d9358
--- /dev/null
+++ b/linux_os/guide/system/permissions/mounting/kernel_module_usb-storage_disabled/oval/wrlinux1019.xml
@@ -0,0 +1,88 @@
+<def-group>
+ <!-- THIS FILE IS GENERATED by create_kernel_modules_disabled.py.  DO NOT EDIT.  -->
+  <definition class="compliance"
+  id="kernel_module_usb-storage_disabled" version="1">
+    <metadata>
+      <title>Disable usb-storage Kernel Module</title>
+      <affected family="unix">
+        <platform>Red Hat Enterprise Linux 7</platform>
+        <platform>multi_platform_fedora</platform>
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The kernel module usb-storage should be disabled.</description>
+      <reference source="galford" ref_id="20150819" ref_url="test_attestation"/>
+    </metadata>
+    <criteria operator="OR">
+      <criterion test_ref="test_kernmod_usb-storage_disabled" comment="kernel module usb-storage disabled in /etc/modprobe.d" />
+      <criterion test_ref="test_kernmod_usb-storage_modprobeconf" comment="kernel module usb-storage disabled in /etc/modprobe.conf" />
+      <criterion test_ref="test_kernmod_usb-storage_etcmodules-load" comment="kernel module usb-storage disabled in /etc/modules-load.d" />
+      <criterion test_ref="test_kernmod_usb-storage_runmodules-load" comment="kernel module usb-storage disabled in /run/modules-load.d" />
+      <criterion test_ref="test_kernmod_usb-storage_libmodules-load" comment="kernel module usb-storage disabled in /usr/lib/modules-load.d" />
+    </criteria>
+  </definition>
+
+  <ind:textfilecontent54_test id="test_kernmod_usb-storage_disabled" version="1" check="all"
+  comment="kernel module usb-storage disabled">
+    <ind:object object_ref="obj_kernmod_usb-storage_disabled" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test id="test_kernmod_usb-storage_modprobeconf" version="1" check="all"
+  comment="kernel module usb-storage disabled in /etc/modprobe.conf">
+    <ind:object object_ref="obj_kernmod_usb-storage_modprobeconf" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test id="test_kernmod_usb-storage_etcmodules-load" version="1" check="all"
+  comment="kernel module usb-storage disabled in /etc/modules-load.d">
+    <ind:object object_ref="obj_kernmod_usb-storage_etcmodules-load" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test id="test_kernmod_usb-storage_runmodules-load" version="1" check="all"
+  comment="kernel module usb-storage disabled in /run/modules-load.d">
+    <ind:object object_ref="obj_kernmod_usb-storage_runmodules-load" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_test id="test_kernmod_usb-storage_libmodules-load" version="1" check="all"
+  comment="kernel module usb-storage disabled in /usr/lib/modules-load.d">
+    <ind:object object_ref="obj_kernmod_usb-storage_libmodules-load" />
+  </ind:textfilecontent54_test>
+
+  <ind:textfilecontent54_object id="obj_kernmod_usb-storage_disabled"
+  version="1" comment="kernel module usb-storage disabled">
+    <ind:path>/etc/modprobe.d</ind:path>
+    <ind:filename operation="pattern match">^.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^\s*install\s+usb-storage\s+(/bin/false|/bin/true)$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="obj_kernmod_usb-storage_modprobeconf"
+  version="1" comment="Check deprecated /etc/modprobe.conf for disablement of usb-storage">
+    <ind:filepath>/etc/modprobe.conf</ind:filepath>
+    <ind:pattern operation="pattern match">^\s*install\s+usb-storage\s+(/bin/false|/bin/true)$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="obj_kernmod_usb-storage_etcmodules-load"
+  version="1" comment="kernel module usb-storage disabled in /etc/modules-load.d">
+    <ind:path>/etc/modules-load.d</ind:path>
+    <ind:filename operation="pattern match">^.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^\s*install\s+usb-storage\s+(/bin/false|/bin/true)$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="obj_kernmod_usb-storage_runmodules-load"
+  version="1" comment="kernel module usb-storage disabled in /run/modules-load.d">
+    <ind:path>/run/modules-load.d</ind:path>
+    <ind:filename operation="pattern match">^.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^\s*install\s+usb-storage\s+(/bin/false|/bin/true)$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_object id="obj_kernmod_usb-storage_libmodules-load"
+  version="1" comment="kernel module usb-storage disabled in /usr/lib/modules-load.d">
+    <ind:path>/usr/lib/modules-load.d</ind:path>
+    <ind:filename operation="pattern match">^.*\.conf$</ind:filename>
+    <ind:pattern operation="pattern match">^\s*install\s+usb-storage\s+(/bin/false|/bin/true)$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+</def-group>
-- 
1.9.1

